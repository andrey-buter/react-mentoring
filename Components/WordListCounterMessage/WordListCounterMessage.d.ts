/// <reference types="react" />
interface Props {
    count: number | undefined;
}
declare const WordListCounterMessage: (props: Props) => JSX.Element;
export default WordListCounterMessage;
