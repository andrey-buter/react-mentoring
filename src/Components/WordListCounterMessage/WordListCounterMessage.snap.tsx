import React from 'react';
import renderer from 'react-test-renderer';
import WordListCounterMessage from './WordListCounterMessage';

describe('WordListCounterMessage snapshot', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<WordListCounterMessage count={undefined} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
})
