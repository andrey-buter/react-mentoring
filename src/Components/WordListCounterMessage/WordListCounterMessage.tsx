import React from 'react';

interface Props {
	count: number | undefined
}

const WordListCounterMessage = (props: Props) => {
	const { count } = props;
	let message = 'Loading...';

	if (0 === count) {
		message = 'No Words Found';
	} else
	if (1 === count) {
		message = '1 Word Found';
	} else {
		message = `${count} Words Found`;
	}

	return (
		<div>
			{message}
		</div>
	);
}

export default WordListCounterMessage;
