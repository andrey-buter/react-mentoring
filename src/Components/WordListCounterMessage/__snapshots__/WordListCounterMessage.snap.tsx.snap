// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordListCounterMessage snapshot renders correctly 1`] = `
<div>
  Loading...
</div>
`;
